src/Storage/TSStorageManager.m

```diff
@@ -225,6 +225,11 @@ - (void)setupDatabaseWithSafeBlockingMigrations:(void (^_Nonnull)())safeBlocking
     OWSFailedAttachmentDownloadsJob *failedAttachmentDownloadsMessagesJob =
         [[OWSFailedAttachmentDownloadsJob alloc] initWithStorageManager:self];
     [failedAttachmentDownloadsMessagesJob asyncRegisterDatabaseExtensions];
+
+    // NOTE: [TSDatabaseView asyncRegistrationCompletion] ensures that
+    // kNSNotificationName_DatabaseViewRegistrationComplete is not fired until all
+    // of the async registrations are complete.
+    [TSDatabaseView asyncRegistrationCompletion];
```

Comment:
Rather than just to get all of the "async database registrations" participating in the old completion observation mechanism, I just went for something similar - using an async write to block on completion.